src/mono/mono/metadata/marshal.c

```diff
@@ -5226,39 +5226,126 @@ mono_marshal_get_unsafe_accessor_wrapper (MonoMethod *accessor_method, MonoUnsaf
 	MonoMethod *res;
 	GHashTable *cache;
 	MonoGenericContext *ctx = NULL;
+	MonoGenericContainer *container = NULL;
 	MonoMethod *orig_method = NULL;
 	WrapperInfo *info;
+	/* generic_wrapper == TRUE means we will create a generic wrapper method. */
+	gboolean generic_wrapper = FALSE;
+	/* is_inflated == TRUE means we will inflate a wrapper method before returning. */
+	gboolean is_inflated = FALSE;
+	/* one or both of generic_wrapper or is_inflated might be set, depending on how we're called. */
 
 	if (member_name == NULL && kind != MONO_UNSAFE_ACCESSOR_CTOR)
 		member_name = accessor_method->name;
 
+	// printf("CAME IN: %s (generic = %d, inflated = %d)\n", mono_method_full_name(accessor_method, TRUE), accessor_method->is_generic?1:0, accessor_method->is_inflated?1:0);
+
+	/*
+	 * the method is either a generic method definition, or it might be inflated, not both at
+         * the same time.
+	 */
+	g_assert (!(accessor_method->is_generic && accessor_method->is_inflated));
+
+
+	if (accessor_method->is_inflated) {
+		MonoMethod *declaring = ((MonoMethodInflated*)accessor_method)->declaring;
+		if (declaring->is_generic) {
+			// JIT gets here sometimes.
+			generic_wrapper = TRUE;
+		}
+		is_inflated = TRUE;
+	}
+
+	if (accessor_method->is_generic) {
+		generic_wrapper = TRUE;
+	}
+
+	if (is_inflated) {
```

Comment:
The `is_inflated` appears redundant since `accessor_method` is already passed to the `emit_unsafe_accessor_[field/ctor/method]_wrapper` function.

Can `accessor_method->is_inflated` be used directly instead of the `is_inflated` variable? The idea is to make the wrapper inflated if the accessor method is inflated and remove `gboolean to_be_inflated`.